faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -171,7 +186,45 @@ public void close() {
    * @see com.faunadb.client.query.Language
    */
   public CompletableFuture<Value> query(Expr expr) {
-    return performRequest(json.valueToTree(expr));
+    return query(expr, Optional.empty());
+  }
+
+  /**
+   * Issues a Query to FaunaDB.
+   * <p>
+   * Queries are constructed by the helper methods in the {@link com.faunadb.client.query.Language} class.
+   * <p>
+   * Responses are represented as structured tree where each node is a {@link Value} instance.
+   * {@link Value} instances can be converted to native types. See {@link Value} class for details.
+   *
+   * @param expr the query to be executed.
+   * @param timeout the timeout for the current query. If a query timeout has been set when creating
+   *                the {@link FaunaClient}, the one set here will be used for the scope of this query instead.
+   * @return a {@link CompletableFuture} containing the root node of the response tree.
+   * @see Value
+   * @see com.faunadb.client.query.Language
+   */
+  public CompletableFuture<Value> query(Expr expr, Duration timeout) {
```

Comment:
I think you can just throw an exception case `timeout` is null

Reply:
Yeah, I've considered that... I'm not 100% sure what the best practice to handle `null` inputs is these days: throw an `IllegalArgumentException` vs wrap it within an `Optional`?

I leaned towards wrapping it within an `Optional` since we have the other method that accepts an `Optional<Duration>`, but I'm fine with changing it if you have strong opinions against it!